src/NServiceBus.AcceptanceTests/Recoverability/When_exception_is_treated_as_unrecoverable.cs

```diff
@@ -0,0 +1,90 @@
+namespace NServiceBus.AcceptanceTests.Recoverability
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using AcceptanceTesting.Customization;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_exception_is_treated_as_unrecoverable : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_move_to_defined_error_queue()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<EndpointWithFailingHandler>(b => b
+                    .DoNotFailOnErrorMessages()
+                    .When((session, ctx) => session.SendLocal(new InitiatingMessage
+                    {
+                        Id = ctx.TestRunId
+                    }))
+                )
+                .WithEndpoint<ErrorSpy>()
+                .Done(c => c.MessageMovedToErrorQueue)
+                .Run();
+
+            Assert.IsTrue(context.MessageMovedToErrorQueue);
```

Comment:
we should verify that no retries were invoked e.g. by having something like a handler invocation counter.